core/chains/terra/chain.go

```diff
@@ -67,7 +70,8 @@ func NewChain(db *sqlx.DB, ks keystore.Terra, logCfg pg.LogConfig, eb pg.EventBr
 			}, nil
 		}),
 	}, lggr)
-	ch.txm = terratxm.NewTxm(db, tc, *gpe, dbchain.ID, cfg, ks, lggr, logCfg, eb)
+	ch.txm = terratxm.NewTxm(db, tc, *gpe, ch.id, cfg, ks, lggr, logCfg, eb)
+	ch.balanceMonitor = monitor.NewBalanceMonitor(ch.id, cfg, lggr, ks, ch.Reader)
```

Comment:
Should this be optional, like EVM? e.g. `cfg.BalanceMonitorEnabled()` - if so, default on or off?

Reply:
seems fairly lightweight and beneficial I'm not sure why you'd want to disable it. We don't let you disable any other prom metrics afaik, I'd argue we should just remove the eth flag